src/main/java/mate/academy/springboot/swagger/service/ProductServiceImpl.java

```
@@ -0,0 +1,48 @@
+package mate.academy.springboot.swagger.service;
+
+import java.math.BigDecimal;
+import java.util.List;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.repository.ProductRepository;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+
+    public ProductServiceImpl(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @Override
+    public Product save(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product getById(Long id) {
+        return productRepository.getById(id);
```

let's throw an exception in case product isn't found